ADR 61: Denylist format (#47)
decentraland/adr: docs/ADR-61-denylist-formatting.md

# Denylist format

## Context and Problem Statement
In the context of making sure that platform services don't serve unwanted content (i.e. copyright, illegal, etc.) a good solution is to implement a denylist of entities, content and addresses. Moreover, taking into account this [proposal](https://governance.decentraland.org/proposal/?id=f68cd110-3e8c-11ec-be0c-afec86cba5e5) from the DAO, there should be a mechanism for the community to have lists of denylisted items in world.

## Proposed solution
The proposed solution for having a denylist is to have a single with the list of denylisted items. The format of that files consists of having one line per denylisted item. If a line starts with a hashtag consider it a comment.
i.e.

```
...

> QmQwJMfhJFeb3LL4NFHXe2Kwam4gUGaCRo9u2sJcRvufWS (entity id)

Reviewer: Is `(entity id)` part of the file or is it just for clarifying purposes?

QmSQm39orj9dpDnK9PheVQX8wWqUB1PSfZaKzfD4X1FfhS (entity id)
QmV6cDFsTmSUFhiZMFNuoiMW9iX5fg9ww1mveGDJrs9evB (entity id)
# a comment
0x89890aF02328Ab6Af9d3D8F0d27A97bb7E10E566 (address)
0xFFDF0bE2aF26B12A4Cb3B7a62a55CeB244C87520 (address)
...
```

### Denylist implementation

If catalysts implement the denylist should impact the following way in the catalysts:

- Catalysts that implement a denylists won't serve any denylisted content or entity. 
- Deployments by denylisted addresses or containing denylisted content won't be accepted
- Catalysts should expose a way to know if a item is denylisted

> 

Reviewer: + catalyst won't synchronize entities with denylisted content?

Author: No, since the deployments with deny listed entities won't be returned in any endpoint

# Denylist format

## Context and Problem Statement

> In the context of making sure that platform services don't serve unwanted content (i.e. copyright, illegal, etc.) a good solution would be to implement a denylist of entities, content and addresses. Moreover, taking into account this [proposal](https://governance.decentraland.org/proposal/?id=f68cd110-3e8c-11ec-be0c-afec86cba5e5) from the DAO, there should be a mechanism for the community to have lists of denylisted items in world.

Reviewer: ```suggestion
In the context of making sure that platform services don't serve unwanted content (i.e. copyright, illegal, etc.) a good solution is to implement a denylist of entities, content and addresses. Moreover, taking into account this [proposal](https://governance.decentraland.org/proposal/?id=f68cd110-3e8c-11ec-be0c-afec86cba5e5) from the DAO, there should be a mechanism for the community to have lists of denylisted items in world.
```

## Proposed solution

> The proposed solution for having a denylist is to have a single file type for denylisting where each line is denylisted item. If a line starts with a hashtag consider it a comment.

Reviewer: ```suggestion
The proposed solution for having a denylist is to have a single with the list of denylisted items. The format of that files consists of having one line per denylisted item. If a line starts with a hashtag consider it a comment.
```

i.e.

```
...
QmQwJMfhJFeb3LL4NFHXe2Kwam4gUGaCRo9u2sJcRvufWS (entity id)
QmSQm39orj9dpDnK9PheVQX8wWqUB1PSfZaKzfD4X1FfhS (entity id)
QmV6cDFsTmSUFhiZMFNuoiMW9iX5fg9ww1mveGDJrs9evB (entity id)
# a comment
0x89890aF02328Ab6Af9d3D8F0d27A97bb7E10E566 (address)
0xFFDF0bE2aF26B12A4Cb3B7a62a55CeB244C87520 (address)
...
```

### Denylist implementation

If catalysts implement the denylist should impact the following way in the catalysts:

- Catalysts that implement a denylists won't serve any denylisted content or entity. 
- Deployments by denylisted addresses or containing denylisted content won't be accepted
- Catalysts should expose a way to know if a item is denylisted

Each catalyst could have their own private file for denylisted items.

> There might also be a common denylist file saved in the blockchain that every catalyst in the DAO could be forced to implement, each denylisted item in this file should need to be added via a DAO Poll. 

Reviewer: ```suggestion
There might also be a common denylist file saved in the blockchain that every catalyst in the DAO would be forced to implement, each denylisted item in this file should need to be added via a DAO Poll. 
```

Author: It's could in this case

> There could be a denylist in the client, preventing messages from certain addresses, downloading certain scenes, etc.

Reviewer: Where would this file live?
